### PR TITLE
Allow whitelist for test_dmesg()

### PR DESCRIPTION
**Pull request by douglas-raillard-arm**

### CODE REVIEW

**Comment by douglas-raillard-arm** [comment]
@valschneider This could become the idiomatic pattern for adding test-specific parameters that need to be set by a regular user when running the tests (as opposed as hacking with `exekall run --param` or `--sweep`).

**Comment by douglas-raillard-arm** [comment]
@valschneider This PR can be considered "finished" now and is not RFC anymore.
The commits reflect the "history of the idea", going from the manual implementation based on explicit classes and methods to a completely equivalent and automatic way of doing it.

This reduces the code clutter and allows straightforward use of any key from any configuration class without having to add extra boilerplate first.

File: lisa/tests/base.py [hunk 1]
```diff
@@ -873,7 +889,8 @@ def dmesg_entries(self):
                 if line.strip()
             ]
 
-    def test_dmesg(self, level='warn', facility=None) -> ResultBundle:
+    def test_dmesg(self, level='warn', facility=None, pattern_whitelist:DmesgTestConf.PatternWhitelist=None) -> ResultBundle:
```
Comment (valschneider) [inline lisa/tests/base.py]: Nit: `pattern_ignore` would be more explicit. I was a bit confused at first (i.e. does having a whitelist mean we'll only consider messages matching the pattern, or will we let messages matching the pattern "pass"?).

File: lisa/conf.py [hunk 2]
```diff
@@ -415,6 +447,9 @@ class MultiSrcConfMeta(abc.ABCMeta):
     template with the ``{generated_help}`` placeholder replaced by a snippet of
     ResStructuredText containing the list of allowed keys.
 
+    It will also create the types specified using ``newtype`` in the
+    :class:`KeyDesc`, along with a getter to expose it to ``exekall``.
```
Comment (valschneider) [inline lisa/conf.py]: Might want to document here (or somewhere else - anywhere as long as it's documented!) that it's exposed as a camelcased attribute.

File: lisa/tests/base.py (hunk 1 above)
Comment (douglas-raillard-arm) [reply lisa/tests/base.py]: I'll go for `ignore_patterns` so it shows that it's a list (or at least an iterable).

File: lisa/conf.py (hunk 2 above)
Comment (douglas-raillard-arm) [reply lisa/conf.py]: It was shortly documented in KeyDesc docstring, but is now more explicit.

File: lisa/tests/base.py [hunk 3]
```diff
@@ -886,16 +902,32 @@ def test_dmesg(self, level='warn', facility=None) -> ResultBundle:
...
+        logger = self.get_logger()
+
+        if ignored_patterns:
+            logger.info('Will ignore patterns in dmesg output: {}'.format(ignored_patterns))
+            regex_whitelist = [
+                re.compile(pattern)
+                for pattern in ignored_patterns
+            ]
+        else:
+            regex_whitelist = []
```
Comment (valschneider) [inline lisa/tests/base.py]: Nit: might want to rename that to `regex_ignore` for consistency. Otherwise LGTM.

**Pull request merged**